src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyState = { ...state };
+  const massiveToReturn = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(copyState, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete copyState[key];
+      }
+    }
+
+    if (action.type === 'clear') {
+      for (const prop in copyState) {
+        delete copyState[prop];
+      }
+    }
+    massiveToReturn.push({ ...copyState });
```

Comment:
[CODE STYLE]: Instead of using multiple `if` statements, use a `switch` statement to handle the different `action.type` values. This will make the code more readable and maintainable. Also, ensure to include a `default` case to handle any unexpected `action.type` values.